validator/sawtooth_validator/networking/handlers.py

```diff
@@ -82,21 +83,31 @@ def handle(self, connection_id, message_content):
         # Need to use join here to get the string "0.0.0.0". Otherwise,
         # bandit thinks we are binding to all interfaces and returns a
         # Medium security risk.
+        error = False
         interfaces = ["*", ".".join(["0", "0", "0", "0"])]
         interfaces += netifaces.interfaces()
-        for interface in interfaces:
-            if interface in message.endpoint:
-                LOGGER.debug("Endpoint cannot include '%s': %s",
-                             interface,
-                             message.endpoint)
-
-                connection_response = ConnectionResponse(
-                    status=ConnectionResponse.ERROR)
-                return HandlerResult(
-                    HandlerStatus.RETURN_AND_CLOSE,
-                    message_out=connection_response,
-                    message_type=validator_pb2.Message.
-                    AUTHORIZATION_CONNECTION_RESPONSE)
+        match = re.search(r'\/\/(.*)\:', message.endpoint, re.I)
```

Comment:
It might be good to break the into a function we could write a few unit tests against (as to not get regressions).

If we did that, we might have "if not is_valid_endpoint(....):" instead of the "if error is True" below.